programs/config_api/src/config_instruction.rs

```diff
@@ -30,15 +18,18 @@ pub fn create_account<T: ConfigState>(
     config_account_pubkey: &Pubkey,
     lamports: u64,
     keys: Vec<(Pubkey, bool)>,
-) -> Instruction {
-    let space = T::max_space() + ConfigKeys::serialized_size(keys) as u64;
-    system_instruction::create_account(
-        from_account_pubkey,
-        config_account_pubkey,
-        lamports,
-        space,
-        &id(),
-    )
+) -> Vec<Instruction> {
+    let space = T::max_space() + ConfigKeys::serialized_size(keys);
+    vec![
+        system_instruction::create_account(
+            from_account_pubkey,
+            config_account_pubkey,
+            lamports,
+            space,
+            &id(),
+        ),
+        initialize_account::<T>(config_account_pubkey),
```

Comment:
Just trying to understand what problem this solves. Are you perhaps adding a new config usage that has non-zero defaults?

Reply:
here, I was mostly trying to fix a test ;)

I don't know if we can assume that a buffer of all zeros is always safely deserialize-able, and wanted for config data (after create()) to always be deserialize-able

Reply:
Ah, okay. A fair precaution, and I could envision the non-zero default being useful at some point